src/controllers/csv-processor.ts

```diff
@@ -250,3 +315,96 @@ export const processCSV = async (dataset: Dataset, page: number, size: number):
         };
     }
 };
+
+export const processCSVFromBlobStorage = async (
+    dataset: Dataset,
+    page: number,
+    size: number
+): Promise<ViewErrDTO | ViewDTO> => {
+    const blobStoageService = new BlobStorageService();
+    try {
+        const datafiles = await dataset.datafiles;
+        const datafile: Datafile | undefined = datafiles
+            .filter((filterfile: Datafile) => filterfile.draft === true)
+            .sort(
+                (first: Datafile, second: Datafile) =>
+                    new Date(second.creationDate).getTime() - new Date(first.creationDate).getTime()
+            )
+            .shift();
+        if (datafile === undefined || datafile === null) {
+            return {
+                success: false,
+                errors: [
+                    {
+                        field: 'dataset',
+                        message: [
+                            { lang: ENGLISH, message: t('errors.no_datafile', { lng: ENGLISH }) },
+                            { lang: WELSH, message: t('errors.no_datafile', { lng: WELSH }) }
+                        ],
+                        tag: { name: 'erorors.no_datafile', params: {} }
+                    }
+                ],
+                dataset_id: dataset.id
+            };
+        }
+        const buff = await blobStoageService.readFile(`${datafile.id}.csv`);
+
+        const dataArray: Array<Array<string>> = (await parse(buff, {
+            delimiter: ','
+        }).toArray()) as string[][];
+        const csvheaders = dataArray.shift();
```

Comment:
This will come when we reimplement the backend data model and we can capture that information, load it into the database and then get it back out again.  This will be iterated on soon.